### PR TITLE
Use inline SVG icon to avoid flashing

### PR DESCRIPTION
**Pull request by shaoshing**
**`npm install` is required**

This PR use inline SVG icon (via WebPack) to avoid icon flashing in the following scenarios: 

* expand/collapse layer group #3599 
* switch document #2020 

#### Load time
I observe no impact on the load time or initial display time (time until the layer panel is fully displayed).

#### Layer Panle Performance Comparision

Before and After has the same performance by comapring the results of the Layer Panel Performance Test

  | reference   | inline 
------------- | -------------| ----- 
Layer selection | 22ms      | 19ms 
Collapse AB     | 45ms     | 43ms
Expand AB       | 93ms     | 94ms
(numers are rendering time only)

### CODE REVIEW

**Comment by designedbyscience** [comment]
I like this now that we're using WebPack and can still use files for ease, but Web Pack will inline everything for us... instead of just doing this here, could we update SVGIcon for this to work everywhere?

**Comment by shaoshing** [comment]
@designedbyscience this PR is working as you suggested. `SVGIcon` will inline the SVG by default. We can still manage the files in the same way without change, and WebPack will compile the files once they are changed.

**Comment by shaoshing** [comment]
I saw some icons are a little off their original position. need more work to fix them.

**Comment by iwehrman** [comment]
I ran `npm install` and restarted `grunt debug` but I still get this runtime error in the `SVGIcon` render method:
```
Transfer from documents.beforeStartup to documents.initActiveDocument failed: Error: Cannot find module './ico-libraries-cc.svg'.
    at file:///Users/wehrman/Desktop/photoshop-Release_x86_64/Plug-Ins/Spaces/www/build/spaces-design-en.js:33728:42
    at webpackContextResolve (file:///Users/wehrman/Desktop/photoshop-Release_x86_64/Plug-Ins/Spaces/www/build/spaces-design-en.js:33728:90)
    at webpackContext (file:///Users/wehrman/Desktop/photoshop-Release_x86_64/Plug-Ins/Spaces/www/build/spaces-design-en.js:33725:30)
    at React.createClass.render (file:///Users/wehrman/Desktop/photoshop-Release_x86_64/Plug-Ins/Spaces/www/build/spaces-design-en.js:33530:48)
```

File: src/js/jsx/shared/SplitButton.jsx [hunk 1]
```diff
@@ -52,8 +52,7 @@ define(function (require, exports, module) {
             
             var buttonIcon;
             if (this.props.iconId) {
-                buttonIcon = (<SVGIcon viewBox="0 0 16 16" CSSID={this.props.iconId}
-                    iconPath={this.props.iconPath}/>);
```
Comment (iwehrman) [inline src/js/jsx/shared/SplitButton.jsx]: Why don't we need this prop any longer? (Mainly I just don't understand what the use of `CSSID` is versus `iconPath`.)

Comment (shaoshing) [inline src/js/jsx/shared/SplitButton.jsx]: After switching to inline icons, it is no longer needed to refer the icon via its path. Another reason is that `iconPath` seems not in use any more.

**Comment by shaoshing** [comment]
@designedbyscience @iwehrman @placegraphichere I've fixed all known style issues due to the change of SVG. I also checked the icons under different font sizes (base-8 and base-11) and they are all displaying correctly. So, this PR is now ready for merging. Since the `SVGIcon` is now much simpler with a most of the code removed, can you help review my change to make sure that I did not break anything? :) 

BTW, the icons are correct without applying the `viewBox` property, so I am wondering if I should remove the property completely in other components, or leave them unchanged just in case?

**Comment by iwehrman** [comment]
Everything looks great! Please also take a look @designedbyscience and @placegraphichere, but I'm mergin' away.

**Comment by placegraphichere** [comment]
I am excited about this

**Pull request merged**